### PR TITLE
Add support multiple image formats in pictures

### PR DESCRIPTION
**Pull request by ausi**
This pull request adds the possibility to use mutiple image formats and convert them automatically. 

Examples:

```php
// Convert bitmap images to JPEG and GIFs to PNGs
->setFormats([
    'bmp' => ['jpg'],
    'gif' => ['png'],
])

// Convert JPEG and WEBP images to one another to save bandwith on modern browsers
->setFormats([
    'jpg' => ['webp', 'jpg'],
    'webp' => ['webp', 'jpg'],
])
```

Related: avalanche123/Imagine#718

### CODE REVIEW

File: src/Resizer.php [hunk 1]
```diff
@@ -116,6 +119,11 @@ protected function executeResize(ImageInterface $image, ResizeCoordinatesInterfa
             $imagineOptions['format'] = strtolower(pathinfo($path, PATHINFO_EXTENSION));
         }
 
+        // Fix bug with undefined index notice in Imagine
```
Comment (Toflar) [inline src/Resizer.php]: I guess that's temporary and should be fixed in Imagine?

Comment (ausi) [reply src/Resizer.php]: Yes, pull request is in the works :)

File: src/PictureGenerator.php [hunk 2]
```diff
@@ -49,19 +49,32 @@ public function generate(ImageInterface $image, PictureConfigurationInterface $c
         $this->resizeOptions = clone $options;
         $this->resizeOptions->setTargetPath(null);
 
+        $formats = $this->getFormatsFromConfig(
+            $config,
+            strtolower(pathinfo($image->getPath(), PATHINFO_EXTENSION))
```
[the remark this answers is not in the document]
Comment (ausi) [reply src/PictureGenerator.php]: If this fails, the result should be the empty string and Imagine should itself decide which format to use I think.

[the remark this answers is not in the document]
Comment (Toflar) [reply src/PictureGenerator.php]: Are you sure it doesn't return `false`. PHP returns `false` everywhere :D

[the remark this answers is not in the document]
Comment (ausi) [reply src/PictureGenerator.php]: Yes, I’m sure :)
But I should probably add a unit test for this case...

**Comment by leofeyer** [comment]
What is the use case for this? I understand what the code does, but when would I typically need this?

**Comment by Toflar** [comment]
The use case is that we'll be able to automatically generate e.g. `webp` images for `jpeg` sources eventually giving us the ability to get this:

```html
<picture>
  <source type="image/webp" srcset="assets/images/pic.webp">
  <img src="assets/images/pic.jpg">
</picture>
```

In other words: it's a **major** performance feature as we can let the browser decide if it understands more modern image formats and if so, select this instead of the bigger (in bytes) one.

**Comment by Toflar** [comment]
And it will then make SEO rating tools like PageSpeed and co. very, very happy.

File: src/Resizer.php (hunk 1 above)
Comment (ausi) [reply src/Resizer.php]: https://github.com/avalanche123/Imagine/pull/718

File: src/PictureGenerator.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ausi) [reply src/PictureGenerator.php]: d0260236a0a5605f3b4feaabe3129abfb207342e

**Pull request merged**